esp-hal/src/uart.rs

```diff
@@ -1371,6 +1389,23 @@ impl<'d> Uart<'d, Blocking> {
 
 impl<'d> Uart<'d, Async> {
     /// Reconfigures the driver to operate in [`Blocking`] mode.
+    ///
+    /// ## Example
+    ///
+    /// ```rust, no_run
+    #[doc = crate::before_snippet!()]
+    /// # use esp_hal::uart::{Config, Uart};
+    /// let mut uart1 = Uart::new(
+    ///     peripherals.UART1,
+    ///     Config::default())?
+    /// .with_rx(peripherals.GPIO1)
+    /// .with_tx(peripherals.GPIO2)
+    /// .into_async();
+    ///
+    /// let blocking_uart1 = uart1.into_blocking();
```

Comment:
I don't think we need to explain into_blocking/into_async like this. It's trivial once the concept is explained somewhere.

Reply:
Okay but tokio doesn't have to explain that twice for every struct they expose

Reply:
> Fair enough, in which case we should write this documentation on Blocking and Async and then refer to them for these trivial functions.

+1 since it's a convention and not a trait function this is probably the best we can do - we already have suitable documentation on `Blocking` and `Async` - maybe we want some trivial examples there (but only once)